src/azul/deployment.py

```diff
@@ -434,8 +438,41 @@ def client(self, *args, **kwargs):
         Caching the result of this function is not necessary and will be harmful
         if the cached value is used by a thread other than the one that called
         this function.
+
+        :param azul_logging: whether to log the client's requests and responses
+        """
+        client = self.boto3_session.client(*args, **kwargs)
+        if azul_logging:
+            client.meta.events.register('before-send', self._log_client_request)
+            client.meta.events.register('response-received', self._log_client_response)
+        return client
+
+    def _log_client_request(self,
+                            event_name: str,
+                            request: AWSPreparedRequest,
+                            **kwargs: Any
+                            ) -> Optional[AWSResponse]:
+        """
+        Log the request's AWS API action, HTTP method, URL, and body (unless
+        the body is streamable).
+        """
+        aws_log.info('%s requesting %s to %s', event_name, request.method, request.url)
+        aws_log.info(http_body_log_message(aws_log, 'request', request.body))
+        return None
+
+    def _log_client_response(self,
+                             *,
+                             event_name: str,
+                             **kwargs: Any
+                             ) -> Optional[AWSResponse]:
+        """
+        Log the request's AWS API action and the response's HTTP status and
+        body (unless the body is streamable).
         """
-        return self.boto3_session.client(*args, **kwargs)
+        response = kwargs['response_dict']
+        aws_log.info('Got %s from %s', response['status_code'], event_name)
```

Comment:
```suggestion
        aws_log.info('Got %s from %s', response['status_code'], event_name)
```

What are some example values for event_name? I don't think "from" is the right preposition.

Reply:
Examples are `response-received.dynamodb.GetItem` & `response-received.dynamodb.PutItem`.